packages/library-base/src/nativeDarwin/kotlin/io/realm/kotlin/internal/platform/SystemUtils.kt

```diff
@@ -57,16 +57,16 @@ public actual fun epochInSeconds(): Long =
 @Suppress("MagicNumber")
 internal actual fun currentTime(): RealmInstant {
     val secs = NSDate().timeIntervalSince1970
-    val millis = (secs * 1000 + if (secs > 0) 0.5 else -0.5).toLong()
-    return if (millis < RealmInstant.MIN.epochSeconds * 1000) {
-        RealmInstant.MIN
-    } else if (millis > RealmInstant.MAX.epochSeconds * 1000) {
-        RealmInstant.MAX
-    } else {
-        RealmInstantImpl(
-            millis.floorDiv(1000.toLong()),
-            (millis.mod(1000.toLong()) * 1000).toInt()
-        )
+    return when {
```

Comment:
Not 100% sure exactly what cases we intend to fix with this change? The Dokka written assumes that we are only calling this method with "reasonable" values which should work with the given code?

Reply:
The problem is that the code as it was written would always land here:
```
    } else if (millis > RealmInstant.MAX.epochSeconds * 1000) {
        RealmInstant.MAX // <-------
    } else {
```
because `RealmInstant.MAX` is initialised with `Long.MAX_VALUE`, which when multiplied by `1000` overflows and the result is `0`. Then all calls to `RealmInstant.now()` generate a value for `millis` that is greater than `0` thus the output always is `RealmInstant.MAX` on Darwin.